pytorch_lightning/core/lightning.py

```diff
@@ -1473,6 +1478,43 @@ def untoggle_optimizer(self, optimizer_idx: int):
         # save memory
         self._param_requires_grad_state = {}
 
+    def clip_gradients(
+        self,
+        optimizer: Optimizer,
+        optimizer_idx: int,
+        gradient_clip_val: Union[int, float] = 0.0,
+        gradient_clip_algorithm: str = "norm",
+    ):
+        """Perform Gradient Clipping for the optimizer parameters. Called before :meth:`optimizer_step`.
+
+        Args:
+            optimizer: Current optimizer being used. ``None`` if using manual optimization.
+            optimizer_idx: Index of the current optimizer being used. ``None`` if using manual optimization.
+            gradient_clip_val: The value at which to clip gradients. Passing ``gradient_clip_val=0`` disables gradient
+                clipping.
+            gradient_clip_algorithm: The gradient clipping algorithm to use. Pass ``gradient_clip_algorithm="value"``
+                for clip_by_value, and ``gradient_clip_algorithm="norm"`` for clip_by_norm.
```

Comment:
If it's a hook, the user doesn't call it, yet the docs here say that "passing =0 disables gradient clipping"? 
Meanwhile, this PR also deprecates the trainer arguments, so how is this feature supposed to be used now?